platform/api/openstack/api.go

```diff
@@ -546,6 +546,18 @@ func (a *API) UploadImage(name, path string) (string, error) {
 			return "", fmt.Errorf("web uploading: %w", err)
 		}
 
+		// It usually takes around 10 seconds to extract the image.
+		if err := util.WaitUntilReady(20*time.Second, 5*time.Second, func() (bool, error) {
```

Comment:
I'd be on the safe side and would wait for a minute. :)

Reply:
Ok, let's do this!